src/test/java/org/littil/api/school/api/SchoolResourceTest.java

```diff
@@ -235,78 +288,44 @@ void givenUpdatingNameOfSchoolById_thenShouldUpdateSuccessfully() {
-        School school = createSchool();
-        UUID schoolId = UUID.randomUUID();
-        UUID newSchoolId = UUID.randomUUID();
-
-        school.setId(newSchoolId);
-
-        ErrorResponse errorResponse = given()
-                .contentType(ContentType.JSON)
-                .body(school)
-                .put("/{id}", schoolId)
-                .then()
-                .statusCode(500)
-                .extract().as(ErrorResponse.class);
-
-        assertThat(errorResponse.getErrors())
-                .isNotNull()
-                .hasSize(1)
-                .contains(new ErrorResponse.ErrorMessage("Path variable id does not match School.id"));
-    }
-
-    @Test
-    void givenUpdatingUnknownSchoolById_thenShouldReturnWithErrorResponse() {
-        School school = createSchool();
+    @TestSecurity(user = "littil", roles = "viewer")
+    @OidcSecurity(claims = {
+            @Claim(key = "https://littil.org/littil_user_id", value = "0ea41f01-cead-4309-871c-c029c1fe19bf") })
+    void givenUpdatingUnknownSchool_thenShouldReturnWithErrorResponse() {
+        School school = getDefaultSchool();
         school.setId(UUID.randomUUID());
 
         given()
                 .contentType(ContentType.JSON)
                 .body(school)
-                .put("/{id}", school.getId())
+                .put()
                 .then()
                 .statusCode(404);
     }
 
-    @Test
-    void givenUpdatingByIdWherePayloadIdDeviatesFromPathId_thenShouldReturnWithErrorResponse() {
-        School school = createSchool();
-        school.setId(UUID.randomUUID());
-
-        ErrorResponse errorResponse = given()
-                .contentType(ContentType.JSON)
-                .body(school)
-                .put("/{id}", UUID.randomUUID())
-                .then()
-                .statusCode(500)
-                .extract().as(ErrorResponse.class);
-
-        assertThat(errorResponse.getErrors())
-                .isNotNull()
-                .hasSize(1)
-                .contains(new ErrorResponse.ErrorMessage("Path variable id does not match School.id"));
-    }
-
-    @Test
-    void givenSchoolByIdWithoutIdInPayload_thenShouldReturnWithErrorResponse() {
-        School school = createSchool();
-
-        ErrorResponse errorResponse = given()
+    private School saveSchool(School school) {
+        User createdUser = createAndSaveUser();
+        doReturn(Optional.ofNullable(createdUser)).when(userService).getUserById(any(UUID.class)); 
+        
+        Coordinates coordinates = Coordinates.builder() //
+                .lat(0.0) //
+                .lon(0.0) //
+                .build();
+        doReturn(coordinates).when(coordinatesService).getCoordinates(any(), any());
+        
+        doNothing().when(authenticationService).addAuthorization(any(),any(), any());
+          
+        School saved = given()
                 .contentType(ContentType.JSON)
                 .body(school)
```

Comment:
The input of .body() has the type School while the PUT method in SchoolResource takes a SchoolPostResource type. This works for the moment since School and SchoolPostResource have the same fields but this might change in the future.

Reply:
Changed unit test, use SchoolPostResource now.

Reply:
One last remark: line 390 has a validator spy. However, I don't think it is required. The test also passes without this validator.
```
Validator validator = spy(Validator.class);
doReturn(Collections.emptySet()).when(validator).validate(school, School.class);
```